chaudloader/src/hooks/stage1.rs

```diff
@@ -133,6 +138,45 @@ unsafe fn on_game_load(
     }
 }
 
+unsafe fn on_pck_load(
+    sound_engine_class: *mut std::ffi::c_void,
+    pck_file_name: *const winapi::shared::ntdef::WCHAR,
+    out_pck_id: *mut u32
+) -> i32 {
+    let pck_wstr = std::ffi::OsString::from_wide(
+        std::slice::from_raw_parts(pck_file_name,
+            winapi::um::winbase::lstrlenW(pck_file_name) as usize
+        )
+    );
+
+    let return_val = LoadFilePackage.call(sound_engine_class, pck_file_name, out_pck_id);
+    // Only initialize this once in case both Vol1.pck and Vol2.pck are loaded
+    static INITIALIZED: std::sync::atomic::AtomicBool =
+        std::sync::atomic::AtomicBool::new(false);
+    if !INITIALIZED.fetch_or(true, std::sync::atomic::Ordering::SeqCst){
+        if let Some(pck_str) = pck_wstr.to_str() {
+            match pck_str {
+                "Vol1.pck" | "Vol2.pck" => {
```

Comment:
* Shouldn't the `INITIALIZED.fetch_or` go after the `match pck_str`? If the game would, for whatever reason, decide to load a pck other than Vol1.pck or Vol2.pck first, this won't work.
* Comment should be _or_, not _and_?